waku/v2/protocol/waku_store/protocol_metrics.nim

```diff
@@ -0,0 +1,21 @@
+{.push raises: [Defect].}
+
+import metrics
+
+
+declarePublicGauge waku_store_messages, "number of historical messages", ["type"]
```

Comment:
I am extracting this because I cannot declare the same metrics multiple times within the same app (client and server).

IMO, we should have the metrics declared in the same file where they are used. But, if necessary, we'll do it in those protocols.